micro_manager/micro_manager.py

```diff
@@ -437,13 +438,45 @@ def _solve_micro_simulations(self, micro_sims_input: list) -> list:
         micro_sims_output = [None] * self._local_number_of_sims
 
         for count, sim in enumerate(self._micro_sims):
-            start_time = time.time()
-            micro_sims_output[count] = sim.solve(micro_sims_input[count], self._dt)
-            end_time = time.time()
 
-            if self._is_micro_solve_time_required:
+            if not self._crashed_sims[count]:
+                try:
+                    start_time = time.time()
+                    micro_sims_output[count] = sim.solve(
+                        micro_sims_input[count], self._dt)
+                    end_time = time.time()
+                except Exception as error_message:
+                    _, mesh_vertex_coords = self._participant.get_mesh_vertex_ids_and_coordinates(
+                        self._macro_mesh_name)
+                    self._logger.error("Micro simulation at macro coordinates {} has experienced an error. "
+                                       "See next entry for error message. "
+                                       "Keeping values constant at results of previous iteration".format(
+                                           mesh_vertex_coords[count]))
+                    self._logger.error(error_message)
+                    micro_sims_output[count] = self._old_micro_sims_output[count]
+                    self._crashed_sims[count] = True
+            else:
+                micro_sims_output[count] = self._old_micro_sims_output[count]
+            if self._is_micro_solve_time_required and not self._crashed_sims[count]:
                 micro_sims_output[count]["micro_sim_time"] = end_time - start_time
 
+            crash_ratio = np.sum(self._crashed_sims) / len(self._crashed_sims)
+            if crash_ratio > 0.2:
```

Comment:
Comments would help here. In the future we could potentially make this a user configurable parameter.